Fix #18593 - drop db line included in server export if exporting only data

LHofman opened the pull request:
Signed-off-by: Lennert Hofman <hofmanlennert@gmail.com>

### Description

When doing an SQL server export, if only exporting data, don't include the "drop database" line.

Fixes #18593

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

LHofman commented:
I feel like the fix is not in the ideal situation. I would prefer to do it much earlier on so the problem won't come back for future options. Something that checks if only export data is selected that sets all structure options back to the default.

Also, I did add the sign-off to the commit message via rebase, maybe that's why it didn't find it.

fobosplayer committed:
Translated using Weblate (Ukrainian)

Currently translated at 99.9% (3426 of 3427 strings)

[ci skip]

Translation: phpMyAdmin/5.2
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/5-2/uk/
Signed-off-by: Ivan <ivan47044704@gmail.com>

kamil-tekiela commented on src/Plugins/Export/ExportSql.php:
```diff
@@ -803,7 +803,11 @@ public function exportDBCreate(string $db, string $exportType, string $dbAlias =
         $compat = $GLOBALS['sql_compatibility'] ?? 'NONE';
 
-        if (isset($GLOBALS['sql_drop_database'])) {
+        $exportStructure = (
```
Just commenting on code style: the parenthese seem unnecessary and `in_array` should always use strict mode, especially if you compare it to string constants.

LHofman replied:
Can you elaborate on the parentheses being unnecessary?
Would you put this condition on 1 line?

JonekXXX committed:
Translated using Weblate (Polish)

Currently translated at 90.1% (3091 of 3427 strings)

[ci skip]

Translation: phpMyAdmin/5.2
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/5-2/pl/
Signed-off-by: Szymon John <szymonjohnpl@gmail.com>

krisfremen committed:
Translated using Weblate (Macedonian)

Currently translated at 28.8% (987 of 3427 strings)

[ci skip]

Translation: phpMyAdmin/5.2
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/5-2/mk/
Signed-off-by: "Kristijan \"Fremen\" Velkovski" <me@krisfremen.com>

LHofman commented on src/Plugins/Export/ExportSql.php:
```diff
@@ -803,7 +803,11 @@ public function exportDBCreate(string $db, string $exportType, string $dbAlias =
         $compat = $GLOBALS['sql_compatibility'] ?? 'NONE';
 
-        if (isset($GLOBALS['sql_drop_database'])) {
+        $exportStructure = (
+            ! isset($GLOBALS['sql_structure_or_data'])
+            || in_array($GLOBALS['sql_structure_or_data'], ['structure', 'structure_and_data'])
+        );
```
Huh. I haven't seen this syntax used before.

kamil-tekiela replied:
What syntax are you referring to?

LHofman replied:
Putting conditions on multiple lines without parenthesis. But it's fine, I'll follow the syntax

Commit:
Fix #18593 - drop db line included in server export if exporting only data

Signed-off-by: Lennert Hofman <hofmanlennert@gmail.com>

LHofman commented:
I have created a new PR to the correct branch: https://github.com/phpmyadmin/phpmyadmin/pull/19031
